### PR TITLE
Fix varnish config

### PR DESCRIPTION
**Pull request by jewome62**
Fix varnish config because varnish return always MISS

- [x] Add script to test varnish HIT/MISS

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #494
| License       | MIT

### CODE REVIEW

File: docker/varnish/conf/default.vcl [hunk 1]
```diff
@@ -55,10 +67,36 @@ sub vcl_recv {
...
+
   # Don't cache in dev mode
   if (req.url ~ "^/app_dev.php") {
     return(pass);
   }
+
+  return(hash);
+}
+
+# From https://github.com/mattiasgeniar/varnish-4.0-configuration-templates/blob/master/default.vcl
```
Comment (dunglas) [inline docker/varnish/conf/default.vcl]: You can probably just use the one provided in the docs:

```
sub vcl_hash {
    hash_data(req.url);
    if (req.http.host) {
        hash_data(req.http.host);
    } else {
        hash_data(server.ip);
    }
    return (lookup);
}
```

Comment (jewome62) [reply docker/varnish/conf/default.vcl]: Done.
Test added

File: .travis.yml [hunk 2]
```diff
@@ -5,5 +5,8 @@ services:
 
 script:
   - docker-compose build
-  - docker-compose run --rm app bin/console security:check
-  - docker-compose run --rm app bin/console doctrine:schema:create
+  - docker-compose up -d
+  - sleep 30
+  - docker-compose exec app bin/console security:check
+  - docker-compose exec app bin/console doctrine:schema:create
+  - docker-compose exec app docker/varnish/test-varnish.sh
```
Comment (dunglas) [inline .travis.yml]: Can you add a blank line at the end please?

File: docker/varnish/conf/default.vcl [hunk 3]
```diff
@@ -34,6 +34,11 @@ sub vcl_deliver {
   unset resp.http.url;
   # Uncomment the following line to NOT send the "Cache-Tags" header to the client (prevent using CloudFlare cache tags)
   #unset resp.http.Cache-Tags;
+
+  # Add debug header to see if it's the number of hits (if 0 is MISS)
```
Comment (dunglas) [inline docker/varnish/conf/default.vcl]: `# Add a debug to see the number of HITS (0 means MISS)`

File: docker/varnish/test-varnish.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+thirdCall=`curl -X GET "http://varnish/foos" --silent -H  "accept: application/ld+json" -I | grep -Fi "ApiPlatform-Cache-Hits: 2"`
+if [ -z "$thirdCall" ]
+    then echo 'Third call must be HIT 2'; exit 1
+fi
+
+curl -X BAN "http://varnish/" --silent -H "ApiPlatform-Ban-Regex: (^|\,)/foos($|\,)" > /dev/null
+newCall=`curl -X GET "http://varnish/foos" --silent -H  "accept: application/ld+json" -I | grep -Fi "ApiPlatform-Cache-Hits: 0"`
+if [ -z "$newCall" ]
+    then echo 'New call after BAN must be MISS'; exit 1
+fi
```
Comment (dunglas) [inline docker/varnish/test-varnish.sh]: Blank line

File: docker/varnish/test-varnish.sh [hunk 5]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env sh
```
Comment (dunglas) [inline docker/varnish/test-varnish.sh]: Can you inline this file in `.travis.yml` to not pollute the user's skeleton with tests that will not work for him?

**Comment by dunglas** [comment]
Thanks @jewome62

**Pull request merged**